packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -182,7 +182,7 @@ async function getIslandContext (event: H3Event): Promise<NuxtIslandContext> {
   const context = event.method === 'GET' ? getQuery(event) : await readBody(event)
 
   const ctx: NuxtIslandContext = {
-    url: '/',
+    url: event.path, // i.e /__nuxt_island/<component>_<hash>.json, can be ovveriden by user or nuxt island
```

Comment:
To merge, let's keep this limited to the strict fix for now (and discuss/raise new PR)...

For example, by setting url to `/__nuxt_island` vue router will no longer recognise the path and potentially throw errors.

I would suggest a stub router that prevents code from _breaking_ while effectively doing nothing.

```suggestion
    url: '/',
```

Reply:
Sure 👌 what do you think about it @harlan-zw ? This should be enough to fix the issue

Reply:
Sounds good